invoices/invoiceregistry.go

```diff
@@ -1275,7 +1275,11 @@ func (i *InvoiceRegistry) notifyExitHopHtlcLocked(
 			invoiceToExpire = makeInvoiceExpiry(ctx.hash, invoice)
 		}
 
-		i.hodlSubscribe(hodlChan, ctx.circuitKey)
+		// Subscribe to the resolution if the caller specified a
+		// notification channel.
+		if hodlChan != nil {
+			i.hodlSubscribe(hodlChan, ctx.circuitKey)
```

Comment:
what about the notification call when this func. is just called for a lookup, should we also not call them ?

Reply:
not sure I follow the question - if the `holdChan` is not nil then it's called.

Reply:
I mean these calls here `i.notifyHodlSubscribers(htlcFailResolution)` in the same function where we allow now a holdChan of `nil`

Reply:
it is already allowed to be `nil` before this change, and we now make sure `nil` notification is never registered so `i.notifyHodlSubscribers` won't have it.

Reply:
yeah - although we only had a single callsite before and it is not `nil`.